crates/cheatcodes/src/inspector.rs

```diff
@@ -308,6 +379,245 @@ impl Cheatcodes {
+                            gas,
+                        },
+                        created_address,
+                        return_memory_range.clone(),
+                    ));
+                }
+
+                ecx.env.tx.caller = broadcast.new_origin;
+
+                if ecx.journaled_state.depth() == broadcast.depth {
+                    *call_caller = broadcast.new_origin;
+                    let is_fixed_gas_limit = check_if_fixed_gas_limit(ecx, call_gas_limit);
+
+                    let account = &ecx.inner.journaled_state.state()[&broadcast.new_origin];
+                    self.broadcastable_transactions.push_back(BroadcastableTransaction {
+                        rpc: ecx.inner.db.active_fork_url(),
+                        transaction: TransactionRequest {
+                            from: Some(broadcast.new_origin),
+                            to: None,
+                            value: Some(call_value),
+                            input: TransactionInput::new(call_init_code.clone()),
+                            nonce: Some(account.info.nonce),
+                            gas: if is_fixed_gas_limit {
+                                Some(call_gas_limit as u128)
+                            } else {
+                                None
+                            },
+                            ..Default::default()
+                        },
+                    });
+
+                    log_debug_fn(self, call);
+                }
+            }
+        }
+
+        // Allow cheatcodes from the address of the new contract
+        let address = allow_cheatcodes_fn(self, ecx, call);
+
+        // If `recordAccountAccesses` has been called, record the create
+        if let Some(recorded_account_diffs_stack) = &mut self.recorded_account_diffs_stack {
+            recorded_account_diffs_stack.push(vec![AccountAccess {
+                chainInfo: crate::Vm::ChainInfo {
+                    forkId: ecx.db.active_fork_id().unwrap_or_default(),
+                    chainId: U256::from(ecx.env.cfg.chain_id),
+                },
+                accessor: *call_caller,
+                account: address,
+                kind: crate::Vm::AccountAccessKind::Create,
+                initialized: true,
+                oldBalance: U256::ZERO, // updated on (eof)create_end
+                newBalance: U256::ZERO, // updated on (eof)create_end
+                value: call_value,
+                data: call_init_code.clone(),
+                reverted: false,
+                deployedCode: Bytes::new(), // updated on (eof)create_end
+                storageAccesses: vec![],    // updated on (eof)create_end
+                depth: ecx.journaled_state.depth(),
+            }]);
+        }
+
+        None
+    }
+
+    fn end_common<DB>(
+        &mut self,
+        params: EndParams<'_, DB>,
+        create_outcome_fn: impl Fn(InstructionResult, Bytes, Option<Address>) -> CommonEndOutcome,
```

Comment:
I'd prefer if we add this to the where clause and use `F` generic

Reply:
yep, done